370-auth-ui/sdk.md

```diff
@@ -0,0 +1,309 @@
+}
+```
+
+#### Example
+
+##### Example of first party app
+
+```typescript
+import skygear from "@skygear/web";
+skygear.configure({
+  clientID: "client_id",
+  appEndpoint: "https://api.myapp.com",
+  authEndpoint: "https://accounts.myapp.com",
+  assetEndpoint: "https://assets.myapp.com",
+  // isThirdPartyApp: false,
+});
+```
+
+##### Example of third party app
+
+```typescript
+import skygear from "@skygear/web";
+skygear.configure({
+  clientID: "client_id",
+  appEndpoint: "https://api.third-party-app.com",
+  authEndpoint: "https://accounts.first-party-app.com",
+  assetEndpoint: "https://assets.first-party-app.com",
+  isThirdPartyApp: true,
+});
+```
+
+## Authenticating the user
+
+Depending on platform, the flow is a little bit different.
+
+### Native APP SDK
+
+#### API
+
+```typescript
+interface AuthorizeOptions {
+  redirectURI: string;
+  state?: string;
+  onUserDuplicate?: "abort" | "merge" | "create";
+}
+
+// Modeled after https://tools.ietf.org/html/rfc6749#section-5.2
+interface OAuthError {
+  // The state from AuthorizeOptions.
+  state?: string;
+  error: string;
+  error_description?: string;
+  error_uri?: string;
+}
+
+// The promise rejects with an JavaScript Error that looks like a OAuthError.
+function authorize(options: AuthorizeOptions): Promise<User>;
+
+// Open the URL with the user agent that is used to perform authentication.
+// On iOS it is SFAuthenticationSession or ASWebAuthenticationSession
+// On Android it is Chrome Custom Tabs.
+function openURL(url: string): Promise<void>;
+```
+
+#### Example
+
+##### Authenticate in a React Native application
+
+```typescript
+import React, { useCallback } from "react";
+import skygear from "@skygear/react-native";
+import {Button, View} from "react-native";
+function AuthenticationScreen({ navigate } : { navigate: (screenName: string) => void }) {
+  const onPress = useCallback(() => {
+    try {
+      const user = await skygear.auth.authorizeWithWeb({
+        redirectURI: "myappid://anyhost/anypath",
+        endpoint: "https://accounts.myapp.com",
+      });
+      // If the application requires verification, check if the user is verified.
+      if (!user.isVerified) {
+        navigate("VerificationScreen");
+      } else {
+        navigate("HomeScreen");
+      }
+    } catch (e) {
+      // Handle the error properly here.
+      // For example, duplicated user.
+    }
+  }, []);
+  return (
+    <View>
+      <Button title="Sign in" onPress={onPress} />
+    </View>
+  );
+}
+```
+
+##### Open settings in a React Native application
+
+```typescript
+import React, { useCallback } from "react";
+import skygear from "@skygear/react-native";
+import {Button, View} from "react-native";
+function SettingsScreen() {
+  const onPressEditProfile = useCallback(() => {
+    // Application-specific settings
+  }, []);
+  const onPressSecuritySettings = useCallback(() => {
+    skygear.auth.openURL("https://accounts.myapp.com").catch(e => {
```

Comment:
Maybe we should teach developer to go to `https://accounts.myapp.com/settings` instead of `https://accounts.myapp.com`? Although `/` will redirect to `/settings` if user logged in, but it will show non OIDC login page if the Idp session is not valid.